### ISSUE TITLE
Failed to deserialize timestamp with non-UTC time zone

### ISSUE THREAD

**Issue by lgfa29** [hashicorp/nomad-java-sdk#32]
When Nomad API returns a timestamp with timezone different from UTC, the SDK throws an error:

```
com.fasterxml.jackson.databind.exc.InvalidFormatException: Can not construct instance of java.util.Date from String value ("2019-10-17T21:16:25.849112-04:00"): not a valid representation (error: Failed to parse Date value '2019-10-17T21:16:25.849112-04:00': Can not parse date "2019-10-17T21:16:25.849112-0400": while it seems to fit format 'yyyy-MM-dd'T'HH:mm:ss.SSSZ', parsing fails (leniency? null))
```

Full log: https://gist.github.com/lgfa29/369dfd874e5346a89fa7c114a0f1b0ca

**Comment by erhwenkuo** [comment]
We also encounter this issue while calling 

```java
ServerQueryResponse<List<AllocationListStub>> response = jobsApi.allocations(job.getId());
```

The error message is also regarding to the timestamp format:
```text
Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: Can not deserialize value of type java.util.Date from String "2020-03-15T10:28:05.192414616+08:00": not a valid representation (error: Failed to parse Date value '2020-03-15T10:28:05.192414616+08:00': Can not parse date "2020-03-15T10:28:05.192414616+0800": while it seems to fit format 'yyyy-MM-dd'T'HH:mm:ss.SSSZ', parsing fails (leniency? null))
 at [Source: [{"ClientDescription":"Tasks are running","ClientStatus":"running","CreateIndex":26,"CreateTime":1584239268181953553,"DeploymentStatus":{"Canary":false,"Healthy":true,"ModifyIndex":30,"Timestamp":"2020-03-15T10:28
```

**Comment by cgbaker** [comment]
Hi @erhwenkuo , thanks for the PR. Please accept my apologies for sitting on this for so long. 

Can you add some tests? Specifically, I'd like some codified reassurances that this a) solves the problem, and b) doesn't break for `TZ=UTC`? 

I'm going to go ahead and release `0.11.2.0`, but we'll do a patch release for this.

**Comment by cgbaker** [comment]
closed by #34
